src/components/drawing/index.js

```diff
@@ -1057,38 +1056,32 @@ drawing.setScale = function(element, x, y) {
     return transform;
 };
 
-drawing.setPointGroupScale = function(selection, x, y) {
-    var t, scale, re;
+var SCALE_RE = /\s*sc.*/;
 
-    x = x || 1;
-    y = y || 1;
+drawing.setPointGroupScale = function(selection, xScale, yScale) {
+    xScale = xScale || 1;
+    yScale = yScale || 1;
 
-    if(x === 1 && y === 1) {
-        scale = '';
-    } else {
-        // The same scale transform for every point:
-        scale = ' scale(' + x + ',' + y + ')';
-    }
+    if(!selection) return;
 
-    // A regex to strip any existing scale:
-    re = /\s*sc.*/;
+    // The same scale transform for every point:
+    var scale = (xScale === 1 && yScale === 1) ?
+        '' :
+        ' scale(' + xScale + ',' + yScale + ')';
 
     selection.each(function() {
-        // Get the transform:
-        t = (this.getAttribute('transform') || '').replace(re, '');
+        var t = (this.getAttribute('transform') || '').replace(SCALE_RE, '');
         t += scale;
         t = t.trim();
```

Comment:
I don't know whether the string manipulations here are a meaningful contribution to the performance... but this could all be simplified a bit (`indexOf` & `.substr` instead of a regex & `.replace`, no `.trim`) if we could omit the space before the word `scale`. It seems to work fine in my browsers, though in principle it's not supposed to be allowed: ["The individual transform definitions are separated by whitespace and/or a comma."](https://www.w3.org/TR/SVG/coords.html#TransformAttribute)

Alternatively, we could imagine stashing the pre-scaled transform as perhaps a `data-initialtransform` attribute on each element and avoid all this processing...

Not necessary, just a possibility if you think there would be a benefit.

Reply:
I haven't benchmarked this particular block yet, but I know that `Drawing.setTextPointsScale` is 4-5 times slower than this routine here `Drawing.setPointGroupScale`. 

Spending time on making these routines faster would help improve scroll performance. So yeah, thanks for writing this down. I just opened a new ticket about this -> https://github.com/plotly/plotly.js/issues/2625